### PR TITLE
update flask version to 3.0

### PR DESCRIPTION
**Pull request by dinhlongviolin1**

### CODE REVIEW

File: src/taipy/gui/server.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def __init__(
 
         # set json encoder (for Taipy specific types)
         self._flask.json_provider_class = _TaipyJsonProvider
-        self._flask.json = self._flask.json_provider_class(self._flask)
+        self._flask.json = self._flask.json_provider_class(self._flask)  # type: ignore
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/server.py:98]: good fix 😂

**Review by FredLL-Avaiga — APPROVED** [review]

File: src/taipy/gui/server.py (hunk 1 above)
Comment (dinhlongviolin1) [reply src/taipy/gui/server.py:98]: It was a dumb error in the first place :D it requires "App". But Flask instance inherits "App" so...

Comment (dinhlongviolin1) [reply src/taipy/gui/server.py:98]: Nothing we can do unless we fix mypy 😀

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**